controller/internal/enforcer/envoyproxy/authz/server.go

```diff
@@ -0,0 +1,501 @@
+	}
+
+	// otherwise we are just going to allow it for now
+	zap.L().Debug("ext_authz ingress: Access authorized by network policy", zap.String("puID", s.puID))
+	return &ext_authz_v2.CheckResponse{
+		Status: &rpc.Status{
+			Code: int32(rpc.OK),
+		},
+		HttpResponse: &ext_authz_v2.CheckResponse_OkResponse{
+			OkResponse: &ext_authz_v2.OkHttpResponse{},
+		},
+	}, nil
+}
+
+// egressCheck implements the AuthorizationServer for egress connections
+func (s *Server) egressCheck(ctx context.Context, checkRequest *ext_authz_v2.CheckRequest) (*ext_authz_v2.CheckResponse, error) {
+	zap.L().Debug("ext_authz egress: checkRequest", zap.String("puID", s.puID), zap.String("checkRequest", checkRequest.String()))
+
+	// get the PU context
+	pctxRaw, err := s.puContexts.Get(s.puID)
+	if err != nil {
+		zap.L().Error("ext_authz egress: failed to get PU context", zap.String("puID", s.puID), zap.Error(err))
+		return createDeniedCheckResponse(rpc.INTERNAL, envoy_type.StatusCode_InternalServerError, "failed to get PU context"), nil
+	}
+	pctx, ok := pctxRaw.(*pucontext.PUContext)
+	if !ok {
+		zap.L().Error("ext_authz egress: PU context has the wrong type", zap.String("puID", s.puID), zap.String("puContextType", fmt.Sprintf("%T", pctxRaw)))
+		return createDeniedCheckResponse(rpc.INTERNAL, envoy_type.StatusCode_InternalServerError, "PU context has the wrong type"), nil
+	}
+
+	var sourceAdress string
+	var source, dest *ext_authz_v2.AttributeContext_Peer
+	var httpReq *ext_authz_v2.AttributeContext_HttpRequest
+	attrs := checkRequest.GetAttributes()
+	if attrs != nil {
+		source = attrs.GetSource()
+		dest = attrs.GetDestination()
+
+		if source != nil {
+			if addr := source.GetAddress(); addr != nil {
+				if sockAddr := addr.GetSocketAddress(); sockAddr != nil {
+					sourceAdress = sockAddr.GetAddress()
+				}
+			}
+		}
+
+		if request := attrs.GetRequest(); request != nil {
+			httpReq = request.GetHttp()
+		}
+	}
+
+	flow := newEgressFlowRecord(s.puID, pctx, source, dest, httpReq)
+	// TODO: only collect the flow when this is an external event
+	//defer func() {
+	//	zap.L().Debug("ext_authz egress: collecting flow event", zap.String("puID", s.puID), zap.Any("flow", flow))
+	//	s.collector.CollectFlowEvent(flow)
+	//}()
+
+	observedPolicy, netPolicy := pctx.SearchTxtRules(policy.NewTagStoreFromSlice(pctx.Identity().Tags), false)
+	flow.Action = netPolicy.Action
+	flow.PolicyID = netPolicy.PolicyID
+	flow.ObservedAction = observedPolicy.Action
+	flow.ObservedPolicyID = observedPolicy.PolicyID
+
+	// TODO: we can actually already deny the outgoing request here based on policy
+
+	// build our identity token
+	secrets, unlockSecrets := s.secrets.Secrets()
+	defer unlockSecrets()
+	transmittedKey := secrets.TransmittedKey()
+	token, err := servicetokens.CreateAndSign(
+		sourceAdress,
+		pctx.Identity().Tags,
+		pctx.Scopes(),
+		pctx.ManagementID(),
+		defaultValidity,
+		secrets.EncodingKey(),
+	)
```

Comment:
Just FYI: this will go away with the OAuth PRs .. we will stop transmitting the signed public key and ask the servers to hit the OAUTH JWKS API to retrieve it from the token issuer. This is coming. Just noting it here at the time being.

Reply:
cool :) ... looking forward to that!